### ISSUE TITLE
Nullable state should be tracked across lifted conversions

### ISSUE THREAD

**Issue by cston** [dotnet/roslyn#31500]
No warning should be reported for `y.Value`:
```C#
class Program
{
    static void F1(int? x)
    {
        if (x == null) return;
        var y = ~x;
        _ = y.Value; // warning: nullable value type may be null
    }
}
```

**Comment by jaredpar** [comment]
Is this substantially different from #31500

**Comment by cston** [comment]
Closing as a duplicate of https://github.com/dotnet/roslyn/issues/33330 since that issue also includes binary operators.
